src/lib/comms/include/sol-oic-client.h

```diff
@@ -328,59 +332,52 @@ bool sol_oic_client_get_server_info_by_addr(struct sol_oic_client *client,
- * function is that it uses CoAP non-confirmable packets to make the request.
+ * @return A valid client request on success or @c NULL on errors.
+ */
+struct sol_oic_client_request *sol_oic_client_create_request(sol_coap_method_t method, struct sol_oic_resource *res);
+
+/**
+ * @brief Create an oic client request for an specific @a resource, using a
+ * non-confirmable CoAP packet.
  *
- * @param client An oic client instance.
- * @param res The resource that is going to receive the request.
  * @param method The coap request method as documented in @ref
  *        sol_coap_method_t.
- * @param fill_repr_map A callback to be called to fill the request data.
- *        Parameter @a data is a pointer to user's @a fill_repr_map_data and
- *        @a repr_map is a handler to write data to request packet. Use @ref
- *        sol_oic_map_append() to append data to @a repr_map. If @c NULL, no
- *        data will be added to request.
- * @param fill_repr_map_data User's data to be passed to @a fill_repr_map.
- * @param callback Callback to be called when a response from this request
- *        arrives. Parameter @a response_code is the header response code of
- *        this request, @a cli is the @a client used to perform the
- *        request, @a addr is the address of the server and repr_vec is a
- *        handler to access data from response, using @ref SOL_OIC_MAP_LOOP()
- *        macro. @a data is the user's @a callback_data. When timeout is reached
- *        and no packet has arrived, callback is called with @c NULL @a addr and
- *        @c NULL repr_vec so any clean up can be performed.
- * @param callback_data User's data to be passed to @a callback.
+ * @param res The resource that is going to receive the request.
  *
- * @return True if packet was successfully sent. False otherwise.
+ * @return A valid client request on success or @c NULL on errors.
  */
-bool sol_oic_client_resource_non_confirmable_request(struct sol_oic_client *client, struct sol_oic_resource *res,
-    sol_coap_method_t method,
-    bool (*fill_repr_map)(void *data, struct sol_oic_map_writer *repr_map),
-    void *fill_repr_map_data,
-    void (*callback)(sol_coap_responsecode_t response_code, struct sol_oic_client *cli, const struct sol_network_link_addr *addr,
-    const struct sol_oic_map_reader *repr_vec, void *data),
-    const void *callback_data);
+struct sol_oic_client_request *sol_oic_client_create_non_confirmable_request(sol_coap_method_t method, struct sol_oic_resource *res);
+
+/**
+ * @brief Release memory from a request.
+ *
+ * @param request A pointer to the request to be released.
+ */
+void sol_oic_client_request_free(struct sol_oic_client_request *request);
+
+/**
+ * @brief Get the packet data from a client request.
+ *
+ * @param request The request to retrieve the data.
+ *
+ * @return The packet data from this request.
+ */
+struct sol_oic_map_writer *sol_oic_client_request_get_data(struct sol_oic_client_request *request);
```

Comment:
It was clear to me, but I agree that it may be confusing because data can be a reader or a writer.